lib/TPP/ConvertTppToLoops.cpp

```diff
@@ -22,49 +22,75 @@ using namespace mlir::tpp;
 
 namespace {
 
+static bool isScalarVal(Value val) { return !val.getType().isa<ShapedType>(); }
+
+static SmallVector<Value> getLocalIvs(Value in, Value out, ValueRange ivs,
+                                      Value zero) {
+  assert(in.getType().isa<ShapedType>());
+  assert(out.getType().isa<ShapedType>());
+  auto shapedIn = in.getType().dyn_cast<ShapedType>();
+  auto shapedOut = out.getType().dyn_cast<ShapedType>();
+  assert(shapedOut.getRank() >= shapedIn.getRank());
```

Comment:
that's not a reasonable requirement for a function with the generic name of `getLocalIvs`. There is logic here that isn't explicit in the function name / comments.

Reply:
Will change name and add more comment.